bentoml/cli/bento.py

```diff
@@ -237,3 +245,112 @@ def retrieve(bento, target_dir):
         safe_retrieve(bento_service_bundle_path, target_dir)
 
         click.echo('Service %s artifact directory => %s' % (name, target_dir))
+
+    @cli.command(
+        help='Containerize given Bento into a Docker image',
+        short_help="Containerize given Bento into a Docker image",
+    )
+    @click.argument(
+        "bento", type=click.STRING, callback=parse_bento_tag_callback,
+    )
+    @click.option('--push', is_flag=True)
+    @click.option(
+        '--docker-repository',
+        help="Prepends specified Docker repository to image name.",
+    )
+    @click.option(
+        '-u', '--username', type=click.STRING, required=False,
+    )
+    @click.option(
+        '-p', '--password', type=click.STRING, required=False,
+    )
+    def containerize(bento, push, docker_repository, username, password):
+        """Containerize specified BentoService.
+
+        BENTO is the target BentoService to be containerized, referenced by its name
+        and version in format of name:version. For example: "iris_classifier:v1.2.0"
+
+        `bentoml containerize` command also supports the use of the `latest` tag
+        and will automatically use the last built version of your Bento.
+
+        You can also optionally provide a `--push` flag, which will push the built
+        image to the Docker repository specified by the `--docker-repository`.
+
+        If you would like to push to Docker Hub, `--docker-repository` can just be
+        your Docker Hub username. Otherwise, the tag should include the hostname as
+        well. For example, for Google Container Registry, the `--docker-repository`
+        would be `[HOSTNAME]/[PROJECT-ID]`
+        """
+        name, version = bento.split(':')
+        yatai_client = YataiClient()
+
+        get_bento_result = yatai_client.repository.get(name, version)
+        if get_bento_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                get_bento_result.status
+            )
+            raise CLIException(
+                f'BentoService {name}:{version} not found - '
+                f'{error_code}:{error_message}',
+            )
+
+        if get_bento_result.bento.uri.s3_presigned_url:
+            bento_service_bundle_path = get_bento_result.bento.uri.s3_presigned_url
+        else:
+            bento_service_bundle_path = get_bento_result.bento.uri.uri
+
+        _echo(f"Found Bento: {bento_service_bundle_path}")
+
+        if docker_repository is not None:
+            name = f'{docker_repository}/{name}'
+
+        name, version = _make_bento_name_docker_compatible(name, version)
+        tag = f"{name}:{version}"
+        if tag != bento:
+            _echo(
+                f'Bento name or tag was changed to be Docker compatible. \n'
+                f'"{bento}" -> "{tag}"',
+                CLI_COLOR_WARNING,
+            )
+
+        try:
+            docker_api = docker.APIClient()
```

Comment:
Nitpick: I would initialize the client outside of the `try` block--makes it more clear where the error can be raised. 

If initializing the client can raise an APIError, it makes sense for that to have its own try block, since the error source is different (and you'd want to say so when you raise the CLIException).